postgresql/resource_postgresql_role.go

```diff
@@ -449,15 +449,15 @@ func resourcePostgreSQLRoleReadImpl(c *Client, d *schema.ResourceData) error {
 }
 
 // readSearchPath searches for a search_path entry in the rolconfig array.
-// In case no such value is present, it returns DEFAULT.
+// In case no such value is present, it returns nil.
 func readSearchPath(roleConfig pq.ByteaArray) []string {
 	for _, v := range roleConfig {
 		config := string(v)
 		if strings.HasPrefix(config, roleSearchPathAttr) {
 			return strings.Split(strings.TrimPrefix(config, roleSearchPathAttr+"="), ", ")
 		}
 	}
-	return []string{"DEFAULT"}
+	return nil
```

Comment:
How does one now reset the search path back to default?

Reply:
when terraform see that need to reset it executes function "alterSerachPath".
Please check, I've tested very late :(
I hope your functionality works as expected.